### ISSUE TITLE
Optional Args throw errors due to bug

### ISSUE THREAD

**Issue by moloneymb** [SciSharp/TensorFlow.NET#205]
In the following code a key fetch attempt occurs before the contains check. This will throw a dictionary key not found for optional arguments which have not been set. 

Fixing this means operations like Conv2D no longer have to have the default values for optional arguments hard coded into TensorFlowNet.Core and instead can rely on the default values that are encoded in TensorFlow. 

https://github.com/SciSharp/TensorFlow.NET/blob/46530cf022610a285878dc2413d5633d22e68f9f/src/TensorFlowNET.Core/Operations/OpDefLibrary.cs#L166-L171

For my fix I made the following change. You'll probably want to put an error message in here.

```
                    var key = attr_def.Name;
                    if (attrs.ContainsKey(key))
                    {
                        var value = attrs[key];
                        attr_protos[key] = SetAttrValue(op_def, attr_def, value);
                    } else {
                        // If the attribute is not optional we should throw an error instead of sending a message to Console.WriteLine
                        // TensorFlow will throw a NodeDef exception anyway but we could throw a nicer error
                    }
```

**Comment by moloneymb** [comment]
Fix added to pull request #208

**Comment by Oceania2018** [comment]
Appriciate.
